x-pack/packages/kbn-slo-schema/src/schema/indicators.ts

```diff
@@ -11,26 +11,31 @@ import { allOrAnyString } from './common';
 const kqlQuerySchema = t.string;
 
 const filtersSchema = t.array(
-  t.type({
-    meta: t.partial({
-      alias: t.union([t.string, t.null]),
-      disabled: t.boolean,
-      negate: t.boolean,
-      // controlledBy is there to identify who owns the filter
-      controlledBy: t.string,
-      // allows grouping of filters
-      group: t.string,
-      // index and type are optional only because when you create a new filter, there are no defaults
-      index: t.string,
-      isMultiIndex: t.boolean,
-      type: t.string,
-      key: t.string,
-      field: t.string,
-      params: t.any,
-      value: t.string,
+  t.intersection([
+    t.type({
+      meta: t.partial({
+        alias: t.union([t.string, t.null]),
+        disabled: t.boolean,
+        negate: t.boolean,
+        // controlledBy is there to identify who owns the filter
+        controlledBy: t.string,
+        // allows grouping of filters
+        group: t.string,
+        // index and type are optional only because when you create a new filter, there are no defaults
+        index: t.string,
+        isMultiIndex: t.boolean,
+        type: t.string,
+        key: t.string,
+        field: t.string,
+        params: t.any,
+        value: t.string,
+      }),
+      query: t.record(t.string, t.any),
+    }),
+    t.partial({
+      $state: t.any,
```

Comment:
Is it going to be the same data that we store in the SLO's instance? Was there an issue when this type was not present?

Reply:
we don't store this normall anymore from UI. it's just to be backward compatible.

Reply:
I don't understand what is that for? Can you open an issue stating the problem that this PR is actually solving please?

Reply:
Apparently we were allowing excess keys for a while, resulting in some SLOs created with some `$state: { store: "appState" }`overhead. Then excess keys become not allowed somewhere in 8.16, which breaks any API using the filterSchema when provided with such $state key.

This part of the API is really just a dump of the UI settings, nothing better we can do here. I don't expect customer to use that directly

Reply:
Btw, if this issue was introduced in 8.16, then we don't need to backport it to 8.15:

<img src="https://github.com/user-attachments/assets/5a0503cb-b9f3-484e-92b3-36ea3195f08d" width=400 />